nntrainer/tensor/blas_interface.cpp

```diff
@@ -281,24 +281,28 @@ void sscal(const unsigned int N, const float alpha, _FP16 *X, const int incX) {
 
 static _FP16 snrm2_FP16(const unsigned int N, const _FP16 *X, const int incX) {
   unsigned int incx = abs(incX);
-  _FP16 sum = 0;
+  _FP16 sum;
   _FP16 tmp;
 #if (defined USE__FP16 && USE_NEON)
   if (incX == 1) {
     sum = nntrainer::neon::hnrm2(N, X);
   } else {
+    float sum32 = 0;
     for (unsigned int i = 0; i < N; i++) {
       tmp = X[i * incx];
-      sum += tmp * tmp;
+      sum32 += tmp * tmp;
```

Comment:
just wondering. would having `tmp` a float instead of fp16 make a difference?

Reply:
Definitely yes, for several reasons.

1. You know, fp16 arithmetic can only hold up to 65504, so using fp16 for summing variable(?) is not really a good idea.
It is highly likely to make overflow if N is sufficiently big number.
2. For the same reason, it might cause a underflow as well. For example, it might just skip small decimal values into consideration. As far as I am concerned, even with `__fp16` * `__fp16` multiplication, compiler internally changes them into `float`. Thus, storing that value in `__fp16` variable is also not a clever way I guess. At glance, the effect might seem trivial, but eventually it was causing quite big error when compared to output of fp32-cblas computation.

Reply:
Ah, I misunderstood. Thought you were asking about `sum32`.
But like I mentioned above, the compiler will internally compute them with `float` anyway, so think it won't make huge difference I guess.
Additionally, personally think this is better way i terms of consistency as well. SIMD functions for this computation is implemented is like : "multiply with f16 - and accumulate with f32"